File: app/src/main/java/com/lostincontext/data/location/LocationModel.java
package com.lostincontext.data.location;


import android.support.annotation.NonNull;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.android.gms.maps.model.LatLng;

public class LocationModel {

    public final double latitude;
    public final double longitude;
    public final @NonNull String placeName;

    @JsonIgnore private LatLng latLng;

    @JsonCreator
    public LocationModel(@JsonProperty("placeName") @NonNull String placeName,
                         @JsonProperty("latitude") double latitude,
                         @JsonProperty("longitude") double longitude) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.placeName = placeName;
    }

    public String getPlaceName() {
        return placeName;
    }


    public double getLatitude() {
        return latitude;
    }


    public double getLongitude() {
        return longitude;
    }

    public LatLng getLatLng() {
        return new LatLng(latitude, longitude);
    }
}

File: app/src/main/java/com/lostincontext/data/location/LocationModel.kt
package com.lostincontext.data.location


import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.google.android.gms.maps.model.LatLng

@JsonIgnoreProperties(ignoreUnknown = true)
data class LocationModel(var placeName: String,
                         var latitude: Double,
                         var longitude: Double) {
    @JsonIgnore
    fun getLatLng(): LatLng = LatLng(latitude, longitude)
}

